src/Lecture1.hs

```diff
@@ -104,7 +124,13 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Integer
+strSum str = 
+  {- go one by one through string, accumulating a buffer of elements.
+     upon space, append buffer to result list. should handle floating points that way. -}
+  let nums = map read (words str) :: [Integer] {- or just use words I guess -}
+  in foldr (+) 0 nums
```

Comment:
There's already a standard function `sum` to find the sum of all elements in the list 🙂 
```suggestion
  in sum nums
```